client/components/HomeCarousel.jsx

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import '../styles/homeCarousel.css';
+import $ from 'jquery';
+
+//icons
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChevronRight, faChevronLeft } from '@fortawesome/free-solid-svg-icons';
+import { faHeart as faHeartRegular } from '@fortawesome/free-regular-svg-icons';
+
+
+class HomeCarousel extends React.Component {
+  constructor(props){
+    super(props)
+
+    this.state = {
+      // homes
+      homeCarousel: null,
+      prevButton: null,
+      nextButton: null,
+      id: this.props.id,
+      slideOffset: 0,
+      slidesTotal: this.props.photos.length,
+      slideWidth: 300, //I know, hardcoded, need to find a fix
+      slideMaxX: null,
+      controls: null,
+      //dots
+      indicatorDots: null,
+      dotsOffset: 0,
+      dotsindex: 0,
+      dotsTotal: this.props.photos.length,
+      dotsWidth: 17, //I know, hardcoded, need to find a fix
+      dotsMaxX: null,
+      allDots: null,
+      currDot: 0
+
+    }
+  // home
+  this.moveSlides = this.moveSlides.bind(this);
+  // dots
+  this.moveDots = this.moveDots.bind(this);
+  this.setupDots = this.setupDots.bind(this);
+
+  //click left and right
+  this.nextClick = this.nextClick.bind(this);
+  this.prevClick = this.prevClick.bind(this);
+
+
+  }
+
+  //for photos
+  moveSlides(moveAmount) {
+    this.state.homeCarousel.style.transform = 'translateX(-' + moveAmount + 'px)';
+    console.log(this.state.slidesWidth);
+  };
+
+  // for dots
+  moveDots(moveAmount) {
+    this.state.indicatorDots.style.transform = 'translateX(-' + moveAmount + 'px)';
+  };
+
+  nextClick() {
```

Comment:
You could move these functions to another file if you wanted to. This file is pretty long. @jkwak09